src/Native/System.Security.Cryptography.Native/pal_ssl.cpp

```diff
@@ -170,35 +170,45 @@ extern "C" int32_t CryptoNative_SslSessionReused(SSL* ssl)
     return SSL_session_reused(ssl) == 1;
 }
 
+static bool StringSpanEquals(const char* lhs, const char* rhs, size_t lhsLength)
+{
+    if (lhsLength != strlen(rhs))
```

Comment:
> multiple O(n) strlen calls

N is *very* small.

> maybe we just want to check strlen after strncmp?

Not sure that would necessarily be faster.  As it stands now, we only need to look at one string for most comparisons; the other way around, we'd always need to look at both.  I wouldn't want to do this kind of micro-optimization without some data and a real-world scenario test.